src/globus_sdk/experimental/session_error/variants.py

```diff
@@ -0,0 +1,290 @@
+import typing as t
+
+from .session_error import GlobusSessionError, GlobusSessionErrorAuthorizationParameters
+
+T = t.TypeVar("T", bound="LegacySessionErrorVariant")
+
+
+class LegacySessionErrorVariant:
```

Comment:
Is there a reason that this couldn't extend `GlobusSessionError` (adding new instance variables for each variant) rather than being composable into a `GlobusSessionError`?

Ideally the to_session_error call would be a thing that we could hide from a user as much as possible so that they can simply receive the error and check whether it is a `GlobusSessionError` instead of checking whether it is a `LegacySessionErrorVariant` which allows for conversion into a `GlobusSessionError`.

Reply:
In theory a typical user shouldn't be handling these directly at all, they should be either creating a `GlobusSessionError` directly if they're emitting an error response or they should be using the `utils` methods if they're parsing an error response to see if it's a session error (in which case they get back a proper `GlobusSessionError` or `None` if it doesn't match).

I don't think I can loosen the types in descendant classes, right? Thinking this would be a problem particularly with `authorization_parameters` where the child class would have to support a union of `str` and `list[str]` for several fields.

Reply:
Discussed offline but copying that same note here for historical record:

* I didn't realize there was that type of a distinction between underlying authorization parameters.
* Given that these shouldn't be handled by typical users directly at all, I'd argue that we should make it more explicit in naming that they're internal artifacts. If we ensure it doesn't complicate the external interface of these session errors, I'm happy.